tests/units/DB.php

```diff
@@ -913,4 +913,319 @@ function () {
             }
         )->hasMessage('Unknown type to drop: UNKNOWN');
     }
+
+    protected function fieldValidationPatternProvider(): iterable
+    {
+        $null_values = [
+            'NULL',
+            'null',
+        ];
+
+        $text_values = [
+            'test',
+            '^test',
+            'test$',
+            '^test$',
+        ];
+
+        $positive_integer_values = [
+            '128',
+            '>128',
+            '> 128',
+            '>=128',
+            '>= 128',
+            '<512',
+            '< 512',
+            '<=512',
+            '<= 512',
+            '^16',
+            '56$',
+            '^256$',
+        ];
+
+        $negative_integer_values = [
+            '-1',
+            '^-16',
+            '-56$',
+            '^-256$',
+            '>-128',
+            '> -128',
+            '>=-128',
+            '>= -128',
+            '<-256',
+            '< -256',
+            '<=-256',
+            '<= -256',
+        ];
+
+        $float_values = [
+            '10.5',
+            '<10.5',
+            '< 10.5',
+            '<=10.5',
+            '<= 10.5',
+            '>10.5',
+            '> 10.5',
+            '>=10.5',
+            '>= 10.5',
+        ];
+
+        $relative_datetime_values = [
+            // now + X month
+            '<6',
+            '< 6',
+            '<=6',
+            '<= 6',
+            '>6',
+            '> 6',
+            '>=6',
+            '>= 6',
+
+            // now - X month
+            '<-12',
+            '< -12',
+            '<=-12',
+            '<= -12',
+            '>-12',
+            '> -12',
+            '>=-12',
+            '>= -12',
+        ];
+
+        $date_values = [
+            '2023',
+            '2023-01',
+            '-12-', // all dates in december
+            '04-01',
+
+            /*
+             * All the following values are working with the `contains` operator, but their supports has
+             * probably been introduced for the `is` operator.
+
+            // today
+            'NOW',
+            'TODAY',
+
+            // LAST keyword
+            // is working with any string that will produce a valid result with `strtotime`, e.g. `LAST monday of january 2021`
+            'LASTMONDAY',
+            'LAST MONDAY',
+            'LASTTUESDAY',
+            'LAST TUESDAY',
+            'LASTWEDNESDAY',
+            'LAST WEDNESDAY',
+            'LASTTHURSDAY',
+            'LAST THURSDAY',
+            'LASTFRIDAY',
+            'LAST FRIDAY',
+            'LASTSATURDAY',
+            'LAST SATURDAY',
+            'LASTSUNDAY',
+            'LAST SUNDAY',
+
+            // BEGIN keyword
+            'BEGINMONTH',
+            'BEGINYEAR',
+
+            // END keyword
+            'ENDMONTH',
+            'ENDYEAR',
+
+            // intervals
+            '2DAY',
+            '-2DAY',
+            '4WEEK',
+            '-1WEEK',
+            '3MONTH',
+            '-10MONTH',
+            '6YEAR',
```

Comment:
@orthagh 

Should we support this kind of values in a `contains` search? It could potentially permit to do more precise search than with the limited list proposed with the `is` operator, but this is a bit more complex to define a pattern for it.